src/common/snippets/src/pass/mha_tokenization.cpp

```diff
@@ -16,7 +16,8 @@
 
 namespace {
 bool is_supported_tensor(const ov::descriptor::Tensor& t) {
-    return t.get_partial_shape().is_static() && ov::snippets::utils::one_of(t.get_shape().size(), 3lu, 4lu);
+    return t.get_partial_shape().is_static() && ov::snippets::utils::one_of(t.get_shape().size(), 3lu, 4lu) &&
+           (t.get_element_type() != ov::element::f16);
```

Comment:
The current MHA transformation callback matches on first MatMul [node](https://github.com/liubo-intel/openvino/blob/8b1a672c78add030e83b2c2812accf374a2a1fe5/src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp#L975) and analyzes only the body `MatMul0->Softmax->MatMul1` (without input MatMul0 inputs).
At the moment in general this callback returns `false` in fp16 case since `is_supported_matmul()` [checks](https://github.com/liubo-intel/openvino/blob/8b1a672c78add030e83b2c2812accf374a2a1fe5/src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp#L876) for fp16 case.

But as I can see `MhaTokenization` [doesn't support](https://github.com/liubo-intel/openvino/blob/8b1a672c78add030e83b2c2812accf374a2a1fe5/src/common/snippets/src/pass/mha_tokenization.cpp#L24) `Convert` op at the moment so to check only body `MatMul0->Softmax->MatMul1` - is enough for the check and we can remove this condition from snippets transformation.

Since we analyze not whole body, I asked Bo to add this condition just in case.

But seems like we can remove it

Reply:
so let's currently remove this tensor support limitation?

Reply:
I think so, thank you, Bo!